evm/db/chain.py

```diff
@@ -143,7 +137,10 @@ def persist_header_to_db(self, header):
             rlp.encode(header),
         )
 
-        score = self.get_score(header.parent_hash) + header.difficulty
+        if header.parent_hash == GENESIS_PARENT_HASH:
+            score = header.difficulty
+        else:
+            score = self.get_score(header.parent_hash) + header.difficulty
         self.db.set(
             make_block_hash_to_score_lookup_key(header.hash),
```

Comment:
This is somewhat orthogonal but I just realized that the `get_score(..)` function results in a write operation (setting the score).  This strikes me as un-expected behavior and I'm wondering if (separately from this PR) we should split this into two pieces or do something like `get_score(block_hash, set_missing_scores=True)` so that it can be called without accidentally mutating the database.

Reply:
get_score() does not mutate the db, it simply does a `rlp.decode(self.db.get(...))`. The lines above are part of the `persist_header_to_db()` method

Reply:
🤦‍♂️